utils/io/fileutils/temp.go

```diff
@@ -43,3 +58,53 @@ func RemoveTempDir(dirPath string) error {
 	}
 	return nil
 }
+
+// Create a new temp file named "tempPrefix+timeStamp".
+func CreateTempFile() (*os.File, error) {
+	if tempDirBase == "" {
+		return nil, errorutils.CheckError(errors.New("Temp File cannot be created in an empty base dir."))
+	}
+	timestamp := strconv.FormatInt(time.Now().Unix(), 10)
+	fd, err := ioutil.TempFile(tempDirBase, tempPrefix+"*-"+timestamp+tempFileSuffix)
+	return fd, err
+}
+
+// Old runs/tests may left junk at temp dir.
+// Each temp file/Dir is named with prefix+timestamp, search for all temp files/dirs that match the common prefix and validate their timestamp.
+func CleanOldDirs() error {
+	// Get all files at temp dir
+	files, err := ioutil.ReadDir(tempDirBase)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Search for files/dirs that match the template.
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), tempPrefix) {
+			timeStamp, err := extractTimestamp(file.Name())
+			if err != nil {
+				return err
+			}
+			now := time.Now()
```

Comment:
Better to move this to be out of the loop, to avoid checking time.Now() more than once.